src/main/java/com/commercetools/sync/producttypes/ProductTypeSync.java

```diff
@@ -88,17 +87,18 @@ public ProductTypeSync(@Nonnull final ProductTypeSyncOptions productTypeSyncOpti
     /**
      * Fetches existing {@link ProductType} objects from CTP project that correspond to passed {@code batch}.
      * Having existing product types fetched, {@code batch} is compared and synced with fetched objects by
-     * {@link ProductTypeSync#syncBatch(List, List)} function. When fetching existing product types results in
-     * an empty optional then {@code batch} isn't processed.
+     * {@link ProductTypeSync#syncBatch(Set, Set)} function. When fetching existing product types results in
+     * a {@link ProductTypeSyncStatistics} with initial values {@code 0} of created, updated, failed
+     * then {@code batch} isn't processed.
      *
      * @param batch batch of drafts that need to be synced
-     * @return {@link CompletionStage} of {@link Void} that indicates method progress.
+     * @return {@link CompletionStage} of {@link ProductTypeSyncStatistics} that indicates method progress.
      */
     @Override
     protected CompletionStage<ProductTypeSyncStatistics> processBatch(@Nonnull final List<ProductTypeDraft> batch) {
-        final List<ProductTypeDraft> validProductTypeDrafts = batch.stream()
+        final Set<ProductTypeDraft> validProductTypeDrafts = batch.stream()
```

Comment:
Do you know what happens in the case that there are duplicate ProductTypeDraft elements in the batch list? Do you know if there is a unit/integration test for that case? If not, should we create an issue to address it separately?

Reply:
Yes, I could not see any test cases. It would be nice to open a separate issue for this. And we should consider it before this call because of the batch list created iteratively. It can be duplicated in the sync list. Maybe it's better to check before batch operations. What do u think?

Reply:
> Yes, I could not see any test 

How is that possible that there are no test cases? From a first look, there is always the "normal working" test case, then there are the edgy cases like: lists with duplicates, list with null elements, empty list, etc...

>  It can be duplicated in the sync list. 
Then, what happens in that case? error? What is the behaviour of the sync process then?